nflow-engine/src/main/java/io/nflow/engine/service/WorkflowInstanceService.java

```diff
@@ -50,12 +52,27 @@ public WorkflowInstanceService(WorkflowDefinitionService workflowDefinitionServi
   }
 
   /**
-   * Return the workflow instance matching the given id.
+   * Return the workflow instance matching the given id. Fetches child workflows and current state variables but does not fetch
+   * actions.
    * @param id Workflow instance id.
    * @return The workflow instance, or null if not found.
+   * @deprecated Use getWorkflowInstance(int id, Set&lt;WorkflowInstanceInclude&gt; includes, Long maxActions) instead.
    */
+  @Deprecated
   public WorkflowInstance getWorkflowInstance(int id) {
-    return workflowInstanceDao.getWorkflowInstance(id);
+    return getWorkflowInstance(id, EnumSet.of(WorkflowInstanceInclude.CHILD_WORKFLOW_IDS,
+        WorkflowInstanceInclude.CURRENT_STATE_VARIABLES, WorkflowInstanceInclude.STARTED), null);
+  }
+
+  /**
+   * Return the workflow instance matching the given id.
+   * @param id Workflow instance id.
+   * @param includes Set of properties to be loaded.
+   * @param maxActions Maximum number of actions to be loaded.
+   * @return The workflow instance, or null if not found.
+   */
+  public WorkflowInstance getWorkflowInstance(int id, Set<WorkflowInstanceInclude> includes, Long maxActions) {
```

Comment:
Would a better (or additional) interface be `getWorkflowInstance(int id, Long maxActions, WorkflowInstanceInclude... includes)`.
It would allow callers to not wrap the enums in EnumSet (but most likely would a tiny fraction slower).